modules/claims_api/app/services/claims_api/power_of_attorney_request_service/index.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  module PowerOfAttorneyRequestService
+    class Index
+      def initialize(poa_codes:, page_size:, page_index:, filter: {})
+        @poa_codes = poa_codes
+        @page_size = page_size
+        @page_index = page_index
+        @filter = filter
+      end
+
+      def get_poa_list
+        proc_ids = poa_list.pluck('procID')
+
+        poa_requests = ClaimsApi::PowerOfAttorneyRequest.where(proc_id: proc_ids).select(:id, :proc_id)
```

Comment:
Adding index in https://github.com/department-of-veterans-affairs/vets-api/pull/19945 to optimize this query.